src/lua/init.c

```diff
@@ -869,28 +912,11 @@ tarantool_lua_init(const char *tarantool_bin, int argc, char **argv)
 #if defined(ENABLE_BACKTRACE)
 	backtrace_lua_init();
 #endif /* defined(ENABLE_BACKTRACE) */
-	builtin_modcache_init();
-	lua_getfield(L, LUA_REGISTRYINDEX, "_LOADED");
 	for (const char **s = lua_modules; *s; s += 2) {
 		const char *modname = *s;
 		const char *modsrc = *(s + 1);
-		const char *modfile = lua_pushfstring(L,
-			"@builtin/%s.lua", modname);
-		if (luaL_loadbuffer(L, modsrc, strlen(modsrc), modfile))
-			panic("Error loading Lua module %s...: %s",
-			      modname, lua_tostring(L, -1));
-		lua_pushstring(L, modname);
-		lua_call(L, 1, 1);
-		if (!lua_isnil(L, -1)) {
-			lua_setfield(L, -3, modname); /* package.loaded.modname = t */
-		} else {
-			lua_pop(L, 1); /* nil */
-		}
-		lua_pop(L, 1); /* chunkname */
-		builtin_modcache_put(modname, modsrc);
-
+		luaT_set_module_from_source(L, modname, modsrc);
```

Comment:
Looks like you're loading `internal.minifio` and `internal.loaders` once more here, doesn't it?

Reply:
PEBKAC, I overlooked their deletion from `lua_modules`. Disregard this comment.